src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -180,33 +183,34 @@ public static Checksum Create(Checksum checksum1, Checksum checksum2, Checksum c
 #endif
         }
 
-#if !NET5_0_OR_GREATER
-
-        private static Checksum CreateUsingByteArrays(Checksum checksum1, Checksum checksum2)
+        public static Checksum Create(ReadOnlySpan<Checksum.HashData> hashes)
         {
-            using var bytes = s_twoChecksumByteArrayPool.GetPooledObject();
-
-            var bytesSpan = bytes.Object.AsSpan();
-            checksum1.WriteTo(bytesSpan);
-            checksum2.WriteTo(bytesSpan.Slice(HashSize));
-
-            using var hash = s_incrementalHashPool.GetPooledObject();
-            hash.Object.Initialize();
+#if NET
+            return CreateUsingSpans(hashes);
+#else
+            return CreateUsingByteArrays(hashes);
+#endif
+        }
 
-            hash.Object.TransformBlock(bytes.Object, 0, bytes.Object.Length, null, 0);
+#if !NET5_0_OR_GREATER
```

Comment:
Is there any difference between this an "#if !NET" in Roslyn?

Reply:
yes.  it was this version where the APIs were added taht we depend on.  !NET is for 3 and up i believe.

Reply:
I guess I don't understand why we use both here. If we're not compiling against .netcore v<5, it seems like they're equivalent. If that isn't the case, won't CreateUsingSpans be called in the .net3 case but not be defined?

Reply:
i'll see if i can tweak this.  can i do in a different PR?

Reply:
Definitely no problem being in a different PR. Just seemed confusing to me that this code was using both.